app/models/aaib_report.rb

```diff
@@ -0,0 +1,33 @@
+class AaibReport < Document
+
+  validates :date_of_occurrence, presence: true, date: true, unless: ->(report) {
+    report.report_type == "safety-study" && report.date_of_occurrence.blank?
+  }
+
+  FORMAT_SPECIFIC_FIELDS = [
+    :date_of_occurrence,
+    :aircraft_category,
+    :report_type,
+    :location,
+    :aircraft_type,
+    :registration,
+  ]
+
+  attr_accessor *FORMAT_SPECIFIC_FIELDS
+
+  def initialize(params = {})
+    super(params, FORMAT_SPECIFIC_FIELDS)
```

Comment:
I wonder if there's a way to avoid having to send `FORMAT_SPECIFIC_FIELDS` to the superclass initializer?  Perhaps if `FORMAT_SPECIFIC_FIELDS` was a class method?  

```ruby
class Document
  def initialize(params = {})
    self.class.all_fields).each do |field|
      public_send(:"#{field.to_s}=", params.fetch(field, nil))
    end
  end

  def self.all_fields
    [:id, :title, ... ] # e.g. what COMMON_FIELDS is right now
  end
end

class FooBar < Document
  def self.all_fields
    super + [:my, :extra, :fields]
  end
end
```

I might be missing something about how the current COMMON_FIELDS and FORMAT_SPECIFIC_FIELDS are used, but I reckon there'd be a way of reducing the burden of boilerplate on the subclasses.